### PR TITLE
Reworked v_popcount implementation

### PR DESCRIPTION
**Pull request by terfendail**
### This pullrequest changes
Reworked v_popcount implementation to provide number of bits in a single lane

TODO:
- [x] Perf results

```
allow_multiple_commits=1
force_builders=Linux AVX2,Custom
buildworker:Custom=linux-1,linux-2,linux-4
docker_image:Custom=powerpc64le
```

### CODE REVIEW

**Comment by terfendail** [comment]
<details>
<summary>Performance for SSE2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.039|0.031|1.26|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.255|0.194|1.32|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.037|0.028|1.29|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.238|0.180|1.32|
</details>
<details>
<summary>Performance for SSE3 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.039|0.031|1.26|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.259|0.197|1.32|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.036|0.028|1.29|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.242|0.188|1.29|
</details>
<details>
<summary>Performance for SSE4_2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.021|0.021|0.99|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.155|0.166|0.94|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.015|0.015|1.00|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.100|0.109|0.92|
</details>
<details>
<summary>Performance for AVX2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.012|0.012|0.96|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.159|0.149|1.06|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.010|0.010|1.03|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.094|0.076|1.23|
</details>

File: modules/core/src/stat.simd.hpp [hunk 1]
```diff
@@ -70,16 +70,14 @@ int normHamming(const uchar* a, int n)
...
     {
-        v_uint32x4 t = v_setzero_u32();
-        for(; i <= n - v_uint8x16::nlanes; i += v_uint8x16::nlanes)
-        {
-            t += v_popcount(v_load(a + i));
-        }
-        result += v_reduce_sum(t);
+        v_uint64 t = vx_setzero_u64();
+        for(; i <= n - v_uint8::nlanes; i += v_uint8::nlanes)
+            t += v_popcount(v_reinterpret_as_u64(vx_load(a + i)));
```
Comment (alalek) [inline modules/core/src/stat.simd.hpp]: > u64

Does `u32` slower?

Comment (terfendail) [reply modules/core/src/stat.simd.hpp]: Yes, u64 could use _mm_sad_epu8 intrinsics that produce u64 results(technically it produce 16-bit results but scatter them as u64)

Comment (seiko2plus) [reply modules/core/src/stat.simd.hpp]: using CV_POPCNT_U64 increase performance when SSE4.2 is enabled

**Comment by terfendail** [comment]
<details>
<summary>Performance for SSE2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.039|0.031|1.24|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.255|0.204|1.25|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.037|0.028|1.32|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.238|0.188|1.27|
</details>
<details>
<summary>Performance for SSE3 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.039|0.030|1.29|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.259|0.193|1.34|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.036|0.027|1.34|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.242|0.186|1.30|
</details>
<details>
<summary>Performance for SSE4_2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.021|0.021|1.01|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.155|0.164|0.94|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.015|0.015|1.00|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.100|0.112|0.90|
</details>
<details>
<summary>Performance for AVX2 baseline</summary>

| Performance test | Reference time |  PR time | Speedup |
|------------------|----------------|----------|---------|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.012|0.012|1.00|
|norm2::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.159|0.152|1.04|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 640x480)|0.010|0.010|1.00|
|norm::PerfHamming::(NORM_HAMMING, 8UC1, 1920x1080)|0.094|0.079|1.19|
</details>

**Pull request merged**